### PR TITLE
[video][android] Add fullscreen functionality, improve `requireLinearPlayback` prop

### PR DESCRIPTION
**Pull request by behenate**
# Why

This PR adds activity based fullscreen functionality into `expo-video` for android. Also improves the `requireLinearPlayback` prop, which previously would hide the entire progress bar when enabled.

# How

Used a separate activity for the fullscreen. This might cause some issues down the road, in the worst case scenario we can switch to a `Dialog` based implementation.

Require linear playback was fixed by manually finding the `DefaultTimeBar` view, hiding the scrubber and disabling the touch interactions with the progress bar.

# Test Plan

Tested in BareExpo on Android 14 emulator.

https://github.com/expo/expo/assets/31368152/13137bb5-26b6-4a7f-b758-ffa054340aff

### CODE REVIEW

**Comment by fobos531** [comment]
@behenate 

When PiP is implemented on Android, do you think it should also use this approach with activities? I investigated this and it seems like the most common approach to implement PiP in Android is to display the player in the separate activity, and then declare that specific activity as being able to enter PiP. I hope this is not off-topic, but it looks like a good place to discuss future plans for PiP as well.

**Comment by behenate** [comment]
@fobos531 Hi, I did some experimenting with PiP already and likely we'll use this exact approach. There are some issues to figure out eg. making the transition of entering PiP as smooth as possible. We will also have to consider how to make it work well with react navigation - when exiting PiP we should return to the screen it was opened in.

File: packages/expo-video/android/src/main/java/expo/modules/video/FullscreenActivity.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import android.view.WindowInsets
+import android.view.WindowInsetsController
+import android.widget.ImageButton
+import androidx.media3.ui.PlayerView
+
+@androidx.annotation.OptIn(androidx.media3.common.util.UnstableApi::class)
+class FullscreenActivity : Activity() {
```
Comment (lukmccall) [inline packages/expo-video/android/src/main/java/expo/modules/video/FullscreenActivity.kt:12]:
```suggestion
class FullscreenPlayerActivity : Activity() {
```
I think we should rename it to something more descriptive.

File: packages/expo-video/android/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,2 +1,8 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
+  <application>
+  <activity
+    android:name=".FullscreenActivity"
+    android:configChanges="orientation|keyboardHidden|screenSize"
```
Comment (lukmccall) [inline packages/expo-video/android/src/main/AndroidManifest.xml:5]: Do we need the `configChanges` property?

File: packages/expo-video/android/src/main/res/layout/fullscreen_activity.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  android:layout_height="match_parent"
+  android:background="@android:color/black"
+  tools:context=".FullscreenActivity">
+
+  <androidx.media3.ui.PlayerView
+    android:id="@+id/player_view"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent" />
+
+</FrameLayout>
```
Comment (lukmccall) [inline packages/expo-video/android/src/main/res/layout/fullscreen_activity.xml:15]:
```suggestion
</FrameLayout>

```

**Review by lukmccall — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

File: packages/expo-video/android/src/main/AndroidManifest.xml (hunk 2 above)
Comment (behenate) [reply packages/expo-video/android/src/main/AndroidManifest.xml:5]: Fullscreen seems to work fine without it, although most guides suggested using this property for fullscreen. I guess we can add it back if we encounter any issues.

**Pull request merged**